hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationMonitor.java

```diff
@@ -112,6 +112,9 @@ public class InvocationMonitor implements Consumer<Packet>, StaticMetricsProvide
     private final long heartbeatBroadcastPeriodMillis;
     @Probe(name = OPERATION_METRIC_INVOCATION_MONITOR_INVOCATION_SCAN_PERIOD_MILLIS, unit = MS)
     private final long invocationScanPeriodMillis = SECONDS.toMillis(1);
+    private final long scanForLeftMembersMillis;
+    // read/write only in scheduler thread
+    private long memberLeftAtMillis;
```

Comment:
We want to periodically check if invocation target is left the cluster. However there is one trick that we only do for `invocationTimeout` second after a member is left. This is just here as a small optimization, we would be fine without it. Now lets look at the scenario:

- now=1000 (it's actually some other long but assume 1000 for simplicity), `memberLeftAtMillis`=0, `scanForLeftMembersMillis`=120
- `now < memberLeftAtMillis + scanForLeftMembersMillis` is false
- member 1 leaves
- `memberLeftAtMillis`=now=1000
- `now < memberLeftAtMillis + scanForLeftMembersMillis` is true
- now=20
- member 2 leaves
- `memberLeftAtMillis`=now=1020
- `now < memberLeftAtMillis + scanForLeftMembersMillis` is true and will be true until 1140
- now=1140
- `now < memberLeftAtMillis + scanForLeftMembersMillis` is false and we don't check it anymore

since it's guaranteed that we don't have any invocation to any left member at this point we are fine.